src/output-json.c

```diff
@@ -82,10 +84,18 @@ static const char *TRAFFIC_LABEL_PREFIX = "traffic/label/";
 static size_t traffic_id_prefix_len = 0;
 static size_t traffic_label_prefix_len = 0;
 
+static TAILQ_HEAD(, PluginFileType_) output_types =
+    TAILQ_HEAD_INITIALIZER(output_types);
+
+void RegisterPluginFileType(PluginFileType *plugin)
+{
+    SCLogNotice("Registering JSON file type plugin %s", plugin->name);
+    TAILQ_INSERT_TAIL(&output_types, plugin, entries);
```

Comment:
Suggest we introduce the concept of "plugin priority" ... and order the Q by it.

Reply:
I'm not sure what a priority would do.  Yeah, we have to check for duplicate names, including the internal names.  But since you have to explicitly name something to use, I'm not sure how priorities fit?

Reply:
In the future, there will be open source Suricata plugins for various things and folks will include them without careful analysis (I think).

Might be helpful to introduce the concept of priority at an early stage to handle unintended duplications (multiple output plugins, eg)

Reply:
Wouldn't the name prevent duplications?  If you loaded a "kafka" file type, a second one of the same name would fail to load. You could have "my-kafka-plugin", and "your-kafka-plugin", but to use them you have to explicitly name the one you want to use. Nothing would be implicit.

Reply:
DONE: Duplicate name check as well as not allowing a name to conflict with a built-in (regular, syslog, redis, etc).